File: API/APIMeow/Models/MetaCofrinhoTransacao.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Linq;
using System.Threading.Tasks;

namespace APIMeow.Models
{
    public class MetaCofrinhoTransacao
    {
        [Key]
        public int IdRel { get; set; }
        public int? IdMeta { get; set; }
        public int? IdCofrinho { get; set; }
        [ForeignKey("Transacao")]
        public int IdTransacao { get; set; }
        public Transacao Transacao { get; set; }
    }
}

File: front_meow/lib/models/meta_cofrinho_transacao.dart
class MetaCofrinhoTransacao {
  int idRel;
  int? idMeta;
  int? idCofrinho;
  int idTransacao;

  MetaCofrinhoTransacao({
    required this.idRel,
    this.idMeta,
    this.idCofrinho,
    required this.idTransacao,
  });

  factory MetaCofrinhoTransacao.fromJson(Map<String, dynamic> json) {
    return MetaCofrinhoTransacao(
      idRel: json['idRel'],
      idMeta: json['idMeta'],
      idCofrinho: json['idCofrinho'],
      idTransacao: json['idTransacao'],
    );
  }
}
